### PR TITLE
Save map name / delete map

### PR DESCRIPTION
**Pull request by rebeccaRossRobotics**
This PR adds the following functionality:
- Visual label displaying what SLAM mode you are in
- Visual label displaying which map you are using for localization (only displays when in localization mode)
- Allows you specify the name of the file to save the map to, if none is specified then it saves as map
- Allows you to delete a map (will not let you delete default_map so the code doesn't break, and will not let you delete the map you are using localization)
- Only displays the maps used for the launched SLAM type

This PR relies on the PR#56 in ross_robo

To test use the branch feature/nav_simulation_file one extrm repo.
roslaunch mk3_core simulation_all_modules.launch
roslaunch mk3_nav simulation.launch
roslaunch rr_viz rr_viz.launch

### CODE REVIEW

File: rr_viz/src/rr_viz/widgets/mission_dashboard/mapping/slam_supervisor.py [hunk 1]
```diff
@@ -132,29 +154,45 @@ def waitTillDead_and_execute(self, func):
...
+                eligible_maps.append(_map)
+            elif(self.slam_package == "iris_lama") and ".yaml" in _map:
+                eligible_maps.append(_map)
+            elif(self.slam_package == "rtabmap") and ".db" in _map:
+                eligible_maps.append(_map)
+
+
         for _map in local_maps:
-            if _map not in remote_list:
+            if _map not in eligible_maps:
```
Comment (pepeRossRobotics) [inline rr_viz/src/rr_viz/widgets/mission_dashboard/mapping/slam_supervisor.py:195]: good change

**Comment by msadowski** [comment]
I'm yet to start testing this but thanks a lot for this:

```
To test use the branch feature/nav_simulation_file one extrm repo.
roslaunch mk3_core simulation_all_modules.launch
roslaunch mk3_nav simulation.launch
roslaunch rr_viz rr_viz.launch
```

It's super helpful!

File: rr_viz/src/rr_viz/widgets/mission_dashboard/mapping/slam_supervisor.py [hunk 2]
```diff
@@ -110,12 +129,15 @@ def switchToLocalizationSlot(self):
 
     def runLocalization(self):
         _str = StringRequest()
-        _str.str = self.mapListWidget.currentItem().text().split(".")[0]
+        _str.str = self.mapListWidget.currentItem().text().split(".")[0].strip()
```
Comment (msadowski) [inline rr_viz/src/rr_viz/widgets/mission_dashboard/mapping/slam_supervisor.py]: I think this strip might be causing an issue I'm about to describe (just a gut feel, the issue might be somewhere else actually).

**Review by msadowski — COMMENTED** [review]
In general this seems to work well on both sides. I've noticed one bug though:

If you save a map with some leading spaces (so the name looks something like this: `   test_map`) then if you select the map from the list and try to delete it nothing happens. You can see that the delete was queue in one of the loginfos but the map is actually not removed. 

I've also noticed that we are trimming the extensions in some places and I was wondering if it would be useful to try to keep them.

**Comment by rebeccaRossRobotics** [comment]
@msadowski I've made it so that maps with filenames that have leading or trailing spaces can be deleted. I also made the change you suggested in which it the user specifies a filename with '.' in then it replaces it with '_'

**Comment by msadowski** [comment]
Nice! The changes are good! Everything you've fixed seems to be working OK to me.

We might have a slam-toolbox issue though. Can you try downloading the latest version of slam_toolbox, then create a new map and try to load it?

On my end I seem to be getting this error every time I try to load a new map:

image

Interestingly it was OK for the files I might have created before the slam_toolbox upgrade. If it's indeed a slam_toolbox problem then we will need to create a bug report ASAP

**Comment by rebeccaRossRobotics** [comment]
@msadowski I just tried creating a new map and loading it and its working fine on my pc. I checked I had the lastest slam_toolbox before. 
I've had that error before when i was first doing this bit of work and it seemed to be because the map got corrupted at some point

**Comment by msadowski** [comment]
> @msadowski I just tried creating a new map and loading it and its working fine on my pc. I checked I had the lastest slam_toolbox before.
> I've had that error before when i was first doing this bit of work and it seemed to be because the map got corrupted at some point

Got it. Looks like we can merge all of these changes then. I don't know if you want to use glob for filtering files that I mentioned, if not that's OK just create a Jira issue for this - it might be a good first issue for someone in the future.

**Comment by rebeccaRossRobotics** [comment]
> > @msadowski I just tried creating a new map and loading it and its working fine on my pc. I checked I had the lastest slam_toolbox before.
> > I've had that error before when i was first doing this bit of work and it seemed to be because the map got corrupted at some point
> 
> Got it. Looks like we can merge all of these changes then. I don't know if you want to use glob for filtering files that I mentioned, if not that's OK just create a Jira issue for this - it might be a good first issue for someone in the future.

I've just been trying to get the glob method working but its not quite working. I'll commit it as it is for now an create the jira issue. So just to confirm you're happy for me to merge this PR and the PR in ross_robo?

**Comment by msadowski** [comment]
Yes, happy with the changes! Fine with leaving the refactoring of file fetching as a future issue to solve as well.

**Pull request merged**